lib/cdo/aws/s3.rb

```diff
@@ -93,6 +79,27 @@ def self.public_url(bucket, filename)
       Aws::S3::Object.new(bucket, filename, region: CDO.aws_region).public_url
     end
 
+    # Processes an S3 file, requires a block to be executed after the data has
+    # been downloaded to the temporary file (passed as argument to the block).
+    # @param bucket [String] The S3 buckt name.
+    # @param key [String] The S3 key.
+    # @return The handle to the temporary file.
```

Comment:
I don't think this `@return` is right. The last command is the [unlink](https://ruby-doc.org/stdlib-1.9.3/libdoc/tempfile/rdoc/Tempfile.html#method-i-unlink) on line 99, which deletes the local file. 

`Tempfile.unlink` [appears](https://apidock.com/ruby/v1_9_3_392/Tempfile/unlink) to return nil. In any case I don't think the return of `process_file` as it stands is useful (and it's not used in this code), so I'd remove this line.

Reply:
Updated.